### PR TITLE
Print meaningful message when calling scripts in maven

### LINKED ISSUE #4292: [BUG] no meaningful message is surfaced to maven when binary-dedupe fails
_Originally posted by @gerashegalov in https://github.com/NVIDIA/spark-rapids/issues/4265#issuecomment-986072583_

In some situations when binary-dedupe fails Maven only receives exec exit code via ant, but not the message that is being logged to stderr. It would save time if this message was part of the exception ultimately propagated to Maven

### PR DESCRIPTION
**Pull request by res-life**
Print the error message when calling scripts in maven.
Add resultproperty and errorproperty to save the errorMsg and the exitCode, and then check.
This fixes #4292

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
@gerashegalov 
Because the Linux time command prints to stderr by default,  the error info includes time spent info. 


```
[ERROR] /home/chong/code/spark-rapids/dist/maven-antrun/build-parallel-worlds.xml:131: exec binary-dedupe.sh failed 
......
[ERROR] 
[ERROR] real	0m0.011s
[ERROR] user	0m0.008s
[ERROR] sys	0m0.005s
[ERROR] com/nvidia/spark/ExclusiveModeGpuDiscoveryPlugin.class is not bitwise-identical across shims
```

**Comment by gerashegalov** [comment]
> @gerashegalov Because the Linux time command prints to stderr by default, the error info includes time spent info.
> 
> ```
> [ERROR] /home/chong/code/spark-rapids/dist/maven-antrun/build-parallel-worlds.xml:131: exec binary-dedupe.sh failed 
> ......
> [ERROR] 
> [ERROR] real	0m0.011s
> [ERROR] user	0m0.008s
> [ERROR] sys	0m0.005s
> [ERROR] com/nvidia/spark/ExclusiveModeGpuDiscoveryPlugin.class is not bitwise-identical across shims
> ```

I added these `time`s for diagnostics when we found that binary-dedupe was really slow on macOS but this is now mostly resolved and probably should not be logged to console. Feel free to either remove time wrappers or add `--append --output=<file>` to log timing info to a file.

**Comment by res-life** [comment]
1. Removed time wrappers because the total time is printed.
```
[exec] binary-dedupe completed in 19 seconds
```
2. Added the exit code as part of the message.
3. Reordered the binary-dedupe.sh and check-shims-revisions.sh in the build-parallel-worlds.xml.

**Comment by res-life** [comment]
build

File: dist/maven-antrun/build-parallel-worlds.xml [hunk 1]
```diff
@@ -122,15 +122,34 @@
             <fileset dir="${project.build.directory}/parallel-world" includes="spark*/${shimServiceRsrc}"/>
         </concat>
 
+        <exec executable="${project.basedir}/scripts/binary-dedupe.sh"
```
Comment (gerashegalov) [inline dist/maven-antrun/build-parallel-worlds.xml:125]: What's the rationale of executing binary-dedupe.sh first? Previous iteration of the PR was what we wanted IMO. Now we allow again the build to fail on binary compatibility check whereas it's not the root cause in https://github.com/NVIDIA/spark-rapids/pull/4265#issuecomment-986072583

**Review by gerashegalov — CHANGES_REQUESTED** [review]

File: dist/maven-antrun/build-parallel-worlds.xml (hunk 1 above)
Comment (res-life) [reply dist/maven-antrun/build-parallel-worlds.xml]: Updated

**Comment by res-life** [comment]
build

**Review by gerashegalov — APPROVED** [review]
LGTM

**Review by NvTimLiu — APPROVED** [review]
Looks good to me

**Pull request merged**